x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java

```diff
@@ -335,8 +335,10 @@ private static Function<Page, Page> alignPageToAttributes(List<Attribute> attrs,
             var blocks = new Block[mappedPosition.length];
             for (int i = 0; i < blocks.length; i++) {
                 blocks[i] = p.getBlock(mappedPosition[i]);
+                blocks[i].incRef();
```

Comment:
++. This is more correct than the previous version if the same block is mapped to multiple positions.